x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/analysis/VerifierTests.java

```diff
@@ -58,8 +81,280 @@ public void testMisspelledColumnWithMultipleOptions() {
+        accept(idxr, "foo where date_with_format == '2020-02-02'");
+        accept(idxr, "foo where date_with_format == '2020-02-41'");
+        accept(idxr, "foo where date_with_format == '20200241'");
+
+        accept(idxr, "foo where date_with_multi_format == ''");
+        accept(idxr, "foo where date_with_multi_format == '2020-02-02'");
+        accept(idxr, "foo where date_with_multi_format == '2020-02-41'");
+        accept(idxr, "foo where date_with_multi_format == '20200241'");
+        accept(idxr, "foo where date_with_multi_format == '11:12:13'");
+
+        // Test query against unsupported field type date_nanos
+        assertEquals("1:11: Cannot use field [date_nanos_field] with unsupported type [date_nanos]",
+                error(idxr, "foo where date_nanos_field == ''"));
+    }
+
+    public void testBoolean() {
+        final IndexResolution idxr = loadIndexResolution("mapping-boolean.json");
+        accept(idxr, "foo where boolean_field == true");
+        accept(idxr, "foo where boolean_field == 'bar'");
+        accept(idxr, "foo where boolean_field == 0");
+        accept(idxr, "foo where boolean_field == 123456");
+    }
+
+    public void testBinary() {
+        final IndexResolution idxr = loadIndexResolution("mapping-binary.json");
+        accept(idxr, "foo where blob == ''");
+        accept(idxr, "foo where blob == 'bar'");
+        accept(idxr, "foo where blob == 0");
+        accept(idxr, "foo where blob == 123456");
+    }
+
+    public void testRange() {
+        final IndexResolution idxr = loadIndexResolution("mapping-range.json");
+        assertEquals("1:11: Cannot use field [integer_range_field] with unsupported type [integer_range]",
+                error(idxr, "foo where integer_range_field == ''"));
+        assertEquals("1:11: Cannot use field [float_range_field] with unsupported type [float_range]",
+                error(idxr, "foo where float_range_field == ''"));
+        assertEquals("1:11: Cannot use field [long_range_field] with unsupported type [long_range]",
+                error(idxr, "foo where long_range_field == ''"));
+        assertEquals("1:11: Cannot use field [double_range_field] with unsupported type [double_range]",
+                error(idxr, "foo where double_range_field == ''"));
+        assertEquals("1:11: Cannot use field [date_range_field] with unsupported type [date_range]",
+                error(idxr, "foo where date_range_field == ''"));
+        assertEquals("1:11: Cannot use field [ip_range_field] with unsupported type [ip_range]",
+                error(idxr, "foo where ip_range_field == ''"));
+    }
+
+    public void testObject() {
+        final IndexResolution idxr = loadIndexResolution("mapping-object.json");
+        accept(idxr, "foo where endgame.pid == 0");
+
+        assertEquals("1:11: Unknown column [endgame.pi], did you mean [endgame.pid]?",
+                error(idxr, "foo where endgame.pi == 0"));
+
```

Comment:
Please, remove extra empty line.

Reply:
I kind of liked the one line visual separation between successful and failure cases in here

Reply:
This empty line is at the end of the method, before the closing `}` bracket...

Reply:
will fix